File: posbe/src/main/java/org/posbe/model/Product.java
package org.posbe.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "products")
@Getter
@Setter
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private Double unitPrice; // đơn giá
    private String unit;
    private Double salePrice; // giá bán
}

File: posfe/src/api/product.js
import http from "./http";

const URL = "product";

export const productApi = {
  getAll(page, size) {
    return http.get(`${URL}?page=${page}&size=${size}`);
  },
  create(name, unitPrice, unit, salePrice) {
    return http.post(`${URL}`, {
      name: name,
      unitPrice: unitPrice,
      unit: unit,
      salePrice: salePrice,
    });
  },
  put(id, name, unitPrice, unit, salePrice) {
    return http.put(`${URL}/${id}`, {
      name: name,
      unitPrice: unitPrice,
      unit: unit,
      salePrice: salePrice,
    });
  },
  delete(id) {
    return http.delete(`${URL}/${id}`);
  },
};
